src/bgw/scheduler.c

```diff
@@ -241,13 +253,24 @@ scheduled_bgw_job_transition_state_to(ScheduledBgwJob *sjob, JobState new_state)
 			Assert(!sjob->reserved_worker);
 
 			StartTransactionCommand();
+			BgwJobLockResult res = ts_bgw_job_get_share_lock(sjob->job.fd.id, CurrentMemoryContext);
```

Comment:
mixed declaration and code. But I guess the previous team decided this was now allowed?